### ISSUE TITLE
Install Fails on 2008, 2012 64 bit

### ISSUE THREAD

**Issue by shaikmaqs** [bloomberg/collectdwin#2]
Hi,
Downloaded, Unblocked, gave user full permission and tried to install - fails with error below
"Service 'CollectdWinService (64 bit)' (CollectdWinService (64 bit)) failed to start. Verify that you have sufficient privileges to start system services"

Are there any pre reqs to installing this service?

Thanks,
Shaik

**Comment by shalstea** [comment]
No there are no prerequisites to installing other than you must be an admin.

**Comment by shaikmaqs** [comment]
Doesnt seem to work. I tried this install on 5 machines (2008 and 2012) as local admin, keeps failing with error 
"Service 'CollectdWinService (64 bit)' (CollectdWinService (64 bit)) failed to start. Verify that you have sufficient privileges to start system services"
I also see below Events in event viewer.
"A timeout was reached (30000 milliseconds) while waiting for the CollectdWinService (64 bit) service to connect."
I dont see the collectdwin service under services.msc nor in registry.
Thoughts?

**Comment by yogeswaran** [comment]
1. Are you sure you have 64 bit machine?
2. Did the installer create "C:\Program Files\CollectdWin\" directory with the following files?
   - CollectdWinService.exe
   - CollectdWinService.exe.cfg
   - NLog.dll
   - RabbitMQ.Client.dll
   - types.db
3. Do you see following debug log?
   "C:\ProgramData\CollectdWin\CollectdWin_debug.log", if so do you see any errors in the file?
4. Do you have .NET installed? If so what version?

**Comment by mgravlin** [comment]
I have the exact same issue. To answer your questions from my experience:
1. Yes positive 64-bit
2. The installer creates that directory and even the service, however the error prompt just allows you to Retry which goes in a never-ending loop or Cancel to uninstall. If you Cancel/uninstall that directory and service are removed.
3. Do not see that log even during the time when the Program File folder existed.
4. .NET 4.5.2 (what is the min version required?)

**Comment by shaikmaqs** [comment]
Positive to all points Matt covered.

**Comment by yogeswaran** [comment]
Can you please install via command line to capture the log? Please follow these steps:
1. Download [CollectdWin-x64-v0.5.1.msi](https://github.com/bloomberg/collectdwin/releases/download/v0.5.1/CollectdWin-x64-v0.5.1.msi)
2. Open Windows powershell as admin (right click and run as administrator)
3. Uninstall Collectdwin to make sure there is no previous instance installed, run:
   - msiexec /x CollectdWin-x64-v0.5.1.msi /qn /l\* collectdwin_uninstall.log
4. Install Collectdwin , run:
   - msiexec /i CollectdWin-x64-v0.5.1.msi /qn /l\* collectdwin_install.log

Once you have installed, please review the log files (collectdwin_uninstall.log and collectdwin_install.log) for any errors.

**Comment by cdhunt** [comment]
I have had good experience using http://topshelf-project.com/ in the past for Windows service development. It makes the task of debugging much easier. Not sure if you're aiming for a zero-dependency project, but if not, it's worth looking at.

FWIW, I have installed the current release without problem, myself.

**Comment by shaikmaqs** [comment]
Tried the msiexec commands, same result. Install Fails.
collectdwin_uninstall.log:
MSI (s) (4C:28) [20:38:31:787]: Product: CollectdWinService (64 bit) -- Installation failed.

MSI (s) (4C:28) [20:38:31:787]: Windows Installer installed the product. Product Name: CollectdWinService (64 bit). Product Version: 0.5.1.0. Product Language: 1033. Manufacturer: Bloomberg LP. Installation success or error status: 1605.

This action is only valid for products that are currently installed.
D:\CollectWin\CollectdWin-x64-v0.5.1.msi

collectdwin_install.log:
=== Logging started: 6/21/2015  20:40:53 ===
Action start 20:40:53: INSTALL.
Action start 20:40:53: FindRelatedProducts.
Action ended 20:40:53: FindRelatedProducts. Return value 1.
Action start 20:40:53: ValidateProductID.
Action ended 20:40:53: ValidateProductID. Return value 1.
Action start 20:40:53: CostInitialize.
Action ended 20:40:53: CostInitialize. Return value 1.
Action start 20:40:53: FileCost.
Action ended 20:40:53: FileCost. Return value 1.
Action start 20:40:53: CostFinalize.
Action ended 20:40:53: CostFinalize. Return value 1.
Action start 20:40:53: InstallValidate.
Action ended 20:40:53: InstallValidate. Return value 1.
Action start 20:40:53: RemoveExistingProducts.
Action ended 20:40:53: RemoveExistingProducts. Return value 1.
Action start 20:40:53: InstallInitialize.
Action ended 20:40:54: InstallInitialize. Return value 1.
Action start 20:40:54: ProcessComponents.
Action ended 20:40:54: ProcessComponents. Return value 1.
Action start 20:40:54: UnpublishFeatures.
Action ended 20:40:54: UnpublishFeatures. Return value 1.
Action start 20:40:54: StopServices.
Action ended 20:40:54: StopServices. Return value 1.
Action start 20:40:54: DeleteServices.
Action ended 20:40:54: DeleteServices. Return value 1.
Action start 20:40:54: RemoveFiles.
Action ended 20:40:54: RemoveFiles. Return value 0.
Action start 20:40:54: InstallFiles.
Action ended 20:40:54: InstallFiles. Return value 1.
Action start 20:40:54: InstallServices.
Action ended 20:40:54: InstallServices. Return value 1.
Action start 20:40:54: StartServices.
Action ended 20:40:54: StartServices. Return value 1.
Action start 20:40:54: RegisterUser.
Action ended 20:40:54: RegisterUser. Return value 1.
Action start 20:40:54: RegisterProduct.
Action ended 20:40:54: RegisterProduct. Return value 1.
Action start 20:40:54: PublishFeatures.
Action ended 20:40:54: PublishFeatures. Return value 1.
Action start 20:40:54: PublishProduct.
Action ended 20:40:54: PublishProduct. Return value 1.
Action start 20:40:54: InstallFinalize.
MSI (s) (4C:48) [20:41:26:607]: Product: CollectdWinService (64 bit) -- Error 1920. Service 'CollectdWinService (64 bit)' (CollectdWinService (64 bit)) failed to start.  Verify that you have sufficient privileges to start system services.

Error 1920. Service 'CollectdWinService (64 bit)' (CollectdWinService (64 bit)) failed to start.  Verify that you have sufficient privileges to start system services.
Action ended 20:41:26: InstallFinalize. Return value 3.
Action ended 20:41:26: INSTALL. Return value 3.
Property(S): UpgradeCode = {879AD1B0-4D67-4FAC-BE7B-2958CF71AF73}
Property(S): CollectdWin = C:\Program Files\CollectdWin\
Property(S): ProgramFiles64Folder = C:\Program Files\
Property(S): TARGETDIR = D:\
Property(S): SourceDir = D:\CollectWin\
Property(S): VersionNT = 601
Property(S): Manufacturer = Bloomberg LP
Property(S): ProductCode = {8191F804-A116-4304-AEFE-3E7C5A56A5CB}
Property(S): ProductLanguage = 1033
Property(S): ProductName = CollectdWinService (64 bit)
Property(S): ProductVersion = 0.5.1.0
Property(S): SecureCustomProperties = PREVIOUSFOUND
Property(S): MsiLogFileLocation = D:\CollectWin\collectdwin_install.log
Property(S): PackageCode = {DA53D1AB-C99F-41B5-8355-0B1618AA8D9B}
Property(S): ProductState = -1
Property(S): PackagecodeChanging = 1
Property(S): CURRENTDIRECTORY = D:\CollectWin
Property(S): CLIENTUILEVEL = 3
Property(S): CLIENTPROCESSID = 2592
Property(S): VersionDatabase = 200
Property(S): VersionMsi = 5.00
Property(S): VersionNT64 = 601
Property(S): WindowsBuild = 7601
Property(S): ServicePackLevel = 1
Property(S): ServicePackLevelMinor = 0
Property(S): MsiNTProductType = 3
Property(S): MsiNTSuiteEnterprise = 1
Property(S): WindowsFolder = C:\Windows\
Property(S): WindowsVolume = C:\
Property(S): System64Folder = C:\Windows\system32\
Property(S): SystemFolder = C:\Windows\SysWOW64\
Property(S): RemoteAdminTS = 1
Property(S): TempFolder = C:\Users<userid>\AppData\Local\Temp\
Property(S): ProgramFilesFolder = C:\Program Files (x86)\
Property(S): CommonFilesFolder = C:\Program Files (x86)\Common Files\
Property(S): CommonFiles64Folder = C:\Program Files\Common Files\
Property(S): AppDataFolder = C:\Users<userid>\AppData\Roaming\
Property(S): FavoritesFolder = C:\Users<userid>\Favorites\
Property(S): NetHoodFolder = C:\Users<userid>\AppData\Roaming\Microsoft\Windows\Network Shortcuts\
Property(S): PersonalFolder = C:\Users<userid>\Documents\
Property(S): PrintHoodFolder = C:\Users<userid>\AppData\Roaming\Microsoft\Windows\Printer Shortcuts\
Property(S): RecentFolder = C:\Users<userid>\AppData\Roaming\Microsoft\Windows\Recent\
Property(S): SendToFolder = C:\Users<userid>\AppData\Roaming\Microsoft\Windows\SendTo\
Property(S): TemplateFolder = C:\Users<userid>\AppData\Roaming\Microsoft\Windows\Templates\
Property(S): CommonAppDataFolder = C:\ProgramData\
Property(S): LocalAppDataFolder = C:\Users<userid>\AppData\Local\
Property(S): MyPicturesFolder = C:\Users<userid>\Pictures\
Property(S): AdminToolsFolder = C:\Users<userid>\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Administrative Tools\
Property(S): StartupFolder = C:\Users<userid>\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Startup\
Property(S): ProgramMenuFolder = C:\Users<userid>\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\
Property(S): StartMenuFolder = C:\Users<userid>\AppData\Roaming\Microsoft\Windows\Start Menu\
Property(S): DesktopFolder = C:\Users<userid>\Desktop\
Property(S): FontsFolder = C:\Windows\Fonts\
Property(S): GPTSupport = 1
Property(S): OLEAdvtSupport = 1
Property(S): ShellAdvtSupport = 1
Property(S): MsiAMD64 = 6
Property(S): Msix64 = 6
Property(S): Intel = 6
Property(S): PhysicalMemory = 8192
Property(S): VirtualMemory = 15234
Property(S): AdminUser = 1
Property(S): MsiTrueAdminUser = 1
Property(S): LogonUser = <userid>
Property(S): UserSID = S-1-5-21-606747145-1547161642-839522115-244232
Property(S): UserLanguageID = 1033
Property(S): ComputerName = PPPDC9PRD4X5
Property(S): SystemLanguageID = 1033
Property(S): ScreenX = 1024
Property(S): ScreenY = 768
Property(S): CaptionHeight = 19
Property(S): BorderTop = 1
Property(S): BorderSide = 1
Property(S): TextHeight = 16
Property(S): TextInternalLeading = 3
Property(S): ColorBits = 32
Property(S): TTCSupport = 1
Property(S): Time = 20:41:26
Property(S): Date = 6/21/2015
Property(S): MsiNetAssemblySupport = 4.0.30319.18408
Property(S): MsiWin32AssemblySupport = 6.1.7601.17514
Property(S): RedirectedDllSupport = 2
Property(S): MsiRunningElevated = 1
Property(S): Privileged = 1
Property(S): USERNAME = <companyid>.
Property(S): COMPANYNAME = <companyid>.
Property(S): DATABASE = C:\Windows\Installer\557d0.msi
Property(S): OriginalDatabase = D:\CollectWin\CollectdWin-x64-v0.5.1.msi
Property(S): UILevel = 2
Property(S): ACTION = INSTALL
Property(S): ROOTDRIVE = D:\
Property(S): CostingComplete = 1
Property(S): OutOfDiskSpace = 0
Property(S): OutOfNoRbDiskSpace = 0
Property(S): PrimaryVolumeSpaceAvailable = 0
Property(S): PrimaryVolumeSpaceRequired = 0
Property(S): PrimaryVolumeSpaceRemaining = 0
Property(S): INSTALLLEVEL = 1
Property(S): SOURCEDIR = D:\CollectWin\
Property(S): SourcedirProduct = {8191F804-A116-4304-AEFE-3E7C5A56A5CB}
Property(S): ProductToBeRegistered = 1
MSI (s) (4C:48) [20:41:26:654]: Product: CollectdWinService (64 bit) -- Installation failed.

MSI (s) (4C:48) [20:41:26:654]: Windows Installer installed the product. Product Name: CollectdWinService (64 bit). Product Version: 0.5.1.0. Product Language: 1033. Manufacturer: Bloomberg LP. Installation success or error status: 1603.

=== Logging stopped: 6/21/2015  20:41:26 ===

**Comment by shaikmaqs** [comment]
Please note, Have replaced actual user id with <userid> and company id with <companyid>  above.

**Comment by yogeswaran** [comment]
I have released v0.5.2, in this version installer just installs the service and does not attempt to start the service. Please try this version, installation should succeed. 

Confirm that the service is installed by running:
- Open Windows powershell as admin (right click and run as administrator)
- Get-Service "CollectdWinService (64 bit)"

you should get output like this:

```
c:\> Get-Service "CollectdWinService (64 bit)"
Status   Name               DisplayName
------   ----               -----------
Running  CollectdWinServ... CollectdWinService (64 bit)
```

Once you have confirmed that the service was installed, try this command to start the service:
- net start "CollectdWinService (64 bit)"

If service fails to start, please check if there are any errors reported in:
- Event log
- C:\ProgramData\CollectdWin\CollectdWin_debug.log

**Comment by shaikmaqs** [comment]
Tried the new version, same result. fails.

Status   Name               DisplayName

---

Stopped  CollectdWinServ... CollectdWinService (64 bit)

PS C:\Users\smaqsood> net start "CollectdWinService (64 bit)"
The service is not responding to the control function.

More help is available by typing NET HELPMSG 2186.

Below is the entry from event log,
A timeout was reached (60000 milliseconds) while waiting for the CollectdWinService (64 bit) service to connect.
I dont see Debug log under c:\program files\collectdwin\

I tried increasing teh service timeout value from 30 to 60 seconds, same result.

**Comment by yogeswaran** [comment]
- debug log should be in "C:\ProgramData\CollectdWin\" and NOT in "c:\program files\collectdwin\"
- also you can change the log level by editing nlog config section in config file (C:\Program Files\CollectdWin\CollectdWinService.exe.config), replace "Info" with "Trace"

```
<nlog>
...
<logger name="*" writeTo="default" minlevel="Info" />
...
</nlog>
```

**Comment by shaikmaqs** [comment]
checked under c:\programdata\ and no collectdwin folder found. Enabled trace and tried to start the service, no luck. Service did not start and no collectdwin folder under program data.

**Comment by yogeswaran** [comment]
Please send me your email-id to me ( yramalingam1@bloomberg.net ) so that I can send you an invite to collectdwin channel in Slack. We can chat in slack to troubleshoot this issue.

**Comment by shaikmaqs** [comment]
Sent a mail to your Bloomberg ID.
I see below event under Application logs in event viewer post the service start attempt,

Faulting application name: CollectdWinService.exe, version: 0.5.2.0, time stamp: 0x55881072
Faulting module name: KERNELBASE.dll, version: 6.1.7601.18229, time stamp: 0x51fb1677
Exception code: 0xe0434f4d
Fault offset: 0x000000000000940d
Faulting process id: 0x%9
Faulting application start time: 0x%10
Faulting application path: %11
Faulting module path: %12

**Comment by yogeswaran** [comment]
@shaikmaqs - This issue should be fixed, can you test the latest release [0.5.8](https://github.com/bloomberg/collectdwin/releases/tag/v0.5.8) ?

**Comment by yogeswaran** [comment]
This issue is fixed, if you still see this issue in the new release, please re-open.
